[None][feat] Core Metrics Implementation

hcyezhang opened the pull request:
## Description
Related to https://github.com/NVIDIA/TensorRT-LLM/issues/4926
This PR leverages info from `tllm.Response.Result.RequestPerfMetrics` to generate the following core metrics:
- ttft
- tpot
- e2e time
- request_queue_time
- request_success_count (labeled by finish_reason)

These metrics are exported to a prometheus endpoint mounted to the api server `/prometheus/metrics` (separate from the `/metrics` endpoint used for `IterationStats`)
## Test Coverage
- Sample Requests Looks like:
```
curl http://127.0.0.1:9122/v1/chat/completions     -H "Content-Type: application/json"     -d '{
        "model": "/mnt/modelops/models/Qwen3-0.6B",
        "messages":[{"role": "user", "content": "write a poem about sea"}],
        "max_tokens": 4096,
        "stream": false
    }'
    
curl http://127.0.0.1:9122/v1/chat/completions     -H "Content-Type: application/json"     -d '{
        "model": "/mnt/modelops/models/Qwen3-0.6B",
        "messages":[{"role": "user", "content": "write a poem about sea"}],
        "max_tokens": 4096,
        "stream": true
    }' 
    
 curl http://127.0.0.1:9122/v1/chat/completions     -H "Content-Type: application/json"     -d '{
        "model": "/mnt/modelops/models/Qwen3-0.6B",
        "messages":[{"role": "user", "content": "write a poem about sea"}],
        "max_tokens": 4096,
        "stream": false,
	"best_of": 2,         
        "temperature": 0.7,       
        "top_k": 10,              
        "top_p": 0.9,            
        "n": 2
    }'
```
- TRT workflow
[] no beam search / sampling w/o postprocess_workers
[] beam search + sampling w/o postprocess_workers
[] no beam search / sampling w/ `num_postprocess_workers` = 2
[] beam search + sampling w/ `num_postprocess_workers` = 2
- Pytorch workflow
[] no beam search / sampling w/o postprocess_workers
[] no beam search / sampling w/ `num_postprocess_workers` = 2

## Sample Test Results
`$curl http://localhost:9122/prometheus/metrics`
```
# HELP e2e_request_latency_seconds Histogram of end to end request latency in seconds.
# TYPE e2e_request_latency_seconds histogram
e2e_request_latency_seconds_sum{engine_type="undefined",model_name="undefined"} 2.173774999566376
e2e_request_latency_seconds_bucket{engine_type="undefined",le="0.3",model_name="undefined"} 0.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="0.5",model_name="undefined"} 0.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="0.8",model_name="undefined"} 0.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="1.0",model_name="undefined"} 0.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="1.5",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="2.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="2.5",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="5.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="10.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="15.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="20.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="30.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="40.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="50.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="60.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="120.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="240.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="480.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="960.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="1920.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="7680.0",model_name="undefined"} 2.0
e2e_request_latency_seconds_bucket{engine_type="undefined",le="+Inf",model_name="undefined"} 2.0
e2e_request_latency_seconds_count{engine_type="undefined",model_name="undefined"} 2.0
# HELP time_to_first_token_seconds Histogram of time to first token in seconds.
# TYPE time_to_first_token_seconds histogram
time_to_first_token_seconds_sum{engine_type="undefined",model_name="undefined"} 0.24364100024104118
time_to_first_token_seconds_bucket{engine_type="undefined",le="0.001",model_name="undefined"} 0.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="0.005",model_name="undefined"} 0.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="0.01",model_name="undefined"} 1.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="0.02",model_name="undefined"} 1.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="0.04",model_name="undefined"} 1.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="0.06",model_name="undefined"} 1.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="0.08",model_name="undefined"} 1.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="0.1",model_name="undefined"} 1.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="0.25",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="0.5",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="0.75",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="1.0",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="2.5",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="5.0",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="7.5",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="10.0",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="20.0",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="40.0",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="80.0",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="160.0",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="640.0",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="2560.0",model_name="undefined"} 2.0
time_to_first_token_seconds_bucket{engine_type="undefined",le="+Inf",model_name="undefined"} 2.0
time_to_first_token_seconds_count{engine_type="undefined",model_name="undefined"} 2.0
# HELP time_per_output_token_seconds Histogram of time per output token in seconds.
# TYPE time_per_output_token_seconds histogram
time_per_output_token_seconds_sum{engine_type="undefined",model_name="undefined"} 0.005132794646969996
time_per_output_token_seconds_bucket{engine_type="undefined",le="0.01",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="0.025",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="0.05",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="0.075",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="0.1",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="0.15",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="0.2",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="0.3",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="0.4",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="0.5",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="0.75",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="1.0",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="2.5",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="5.0",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="7.5",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="10.0",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="20.0",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="40.0",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="80.0",model_name="undefined"} 2.0
time_per_output_token_seconds_bucket{engine_type="undefined",le="+Inf",model_name="undefined"} 2.0
time_per_output_token_seconds_count{engine_type="undefined",model_name="undefined"} 2.0
# HELP request_queue_time_seconds Histogram of time spent in WAITING phase for request.
# TYPE request_queue_time_seconds histogram
request_queue_time_seconds_sum{engine_type="undefined",model_name="undefined"} 7.699988782405853e-05
request_queue_time_seconds_bucket{engine_type="undefined",le="0.3",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="0.5",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="0.8",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="1.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="1.5",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="2.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="2.5",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="5.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="10.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="15.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="20.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="30.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="40.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="50.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="60.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="120.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="240.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="480.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="960.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="1920.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="7680.0",model_name="undefined"} 2.0
request_queue_time_seconds_bucket{engine_type="undefined",le="+Inf",model_name="undefined"} 2.0
request_queue_time_seconds_count{engine_type="undefined",model_name="undefined"} 2.0
# HELP request_success_total Count of successfully processed requests.
# TYPE request_success_total counter
request_success_total{engine_type="undefined",finish_reason="stop",model_name="undefined"} 2.0
```


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added support for collecting and exposing Prometheus metrics, including request latency and performance statistics, via a new `/prometheus/metrics` endpoint.
  * Introduced options to return detailed performance metrics with generation results, configurable through API arguments.
  * Enhanced internal metrics tracking for request processing, including time to first token, end-to-end latency, and request queue duration.
  * Enabled propagation and logging of performance metrics throughout the generation and postprocessing pipeline.
  * Added a metrics collector for detailed Prometheus integration with labeled counters and histograms.
  * Included performance metrics support in asynchronous generation calls and result handling.

* **Bug Fixes**
  * None.

* **Chores**
  * Improved metrics infrastructure and integration for better observability and monitoring.
  * Added tests validating Prometheus metrics endpoint and performance metrics exposure.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

hcyezhang committed:
feat: core metric impl

Signed-off-by: Ye Zhang <zhysishu@gmail.com>

rebase in progress; onto 82d03ca97

LinPoly commented:
@Shunkangz seems like this PR is not related with trtllm-4926, it should be a nvbug ticket, can you pls check the title?

Shunkangz commented:
> @Shunkangz seems like this PR is not related with trtllm-4926, it should be a nvbug ticket, can you pls check the title?

Thank you. I remove it.

hcyezhang committed:
fix: pre-commit checks

Signed-off-by: Ye Zhang <zhysishu@gmail.com>

rebase in progress; onto 82d03ca97

LinPoly reviewed (commented):
Leave a few comments for this time, feel free to ping/correct me, will review other parts shortly.

Superjomn commented:
@FrankD412 Can you help review the metric calculations in this PR?

FrankD412 commented on tensorrt_llm/executor/result.py:
```diff
@@ -661,3 +698,28 @@ def _topk_logprobs(logits: torch.Tensor, top_k: int,
+    stat = {}
+    if not req_perf_metrics_dict:
+        return stat
+    ttft = req_perf_metrics_dict.get(RequestEventTiming.FIRST_TOKEN_TIME, 0) - \
```
These calculations seem fine to me. I noticed we default to 0, is there are scenario where we don't expect these keys in the dictionary? If any of these come out to be zero won't we return erroneous results?

FrankD412 commented on tensorrt_llm/metrics/collector.py:
```diff
@@ -0,0 +1,98 @@
+        self.histogram_e2e_time_request = Histogram(
+            name="e2e_request_latency_seconds",
+            documentation="Histogram of end to end request latency in seconds.",
+            buckets=[
```
Just a knit-pick, but should we be statically binning like this? If we end up with a distribution we don't expect isn't this brittle?

hcyezhang replied:
> Just a knit-pick, but should we be statically binning like this? If we end up with a distribution we don't expect isn't this brittle?

hmm.. I agree that having these buckets statically defined isn't ideal. We might end up getting all values in a single bin if something goes really wrong, but honestly I cannot think of a way to "predict" the distribution beforehand(e.g. based on some params/conditions). Maybe adjusting them when we observe unexpected patterns could be the way to go.

Shunkangz commented on tensorrt_llm/serve/openai_server.py:
```diff
@@ -150,6 +152,31 @@ def register_routes(self):
+            registry=registry,
+        ).add().instrument(self.app).expose(self.app)
+        metrics_app = make_asgi_app(registry=registry)
+        metrics_route = Mount("/prometheus/metrics", metrics_app)
```
A trivial question here, what kind of data will be returned if we ping the `/prometheus/metrics` without enabling the `return_perf_metrics`?

hcyezhang replied:
thanks, this is def. non-trivial question. If we ping the `/prometheus/metrics` without enabling `return_perf_metrics`, it returns `200` with empty response body.

hcyezhang committed:
fix: consider multi-response when calculating tpot

Signed-off-by: Ye Zhang <zhysishu@gmail.com>

rebase in progress; onto 82d03ca97

hcyezhang commented on tensorrt_llm/utils/utils.py:
```diff
@@ -0,0 +1,22 @@
+        prometheus_multiproc_dir = tempfile.TemporaryDirectory()
+        os.environ["PROMETHEUS_MULTIPROC_DIR"] = prometheus_multiproc_dir.name
+    logger.info(
+        f"PROMETHEUS_MULTIPROC_DIR: {os.environ['PROMETHEUS_MULTIPROC_DIR']}")
```
no particular reasons. moved the "prometheus init" logic to `tensorrt_llm/_utils.py`

LinPoly reviewed (approved):
The overall logic is clear and I left a few comments for possible improvement.

We still need to:
1. add `promethus` to `requirements.txt`
2. add some tests to ensure correctness and prevent your implementation from being broken in the future, maybe we can add them to a new file, like `_test_openai_promethus.py`, under [this directory](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests/unittest/llmapi/apps), ping me if you have any questions about my comments or about tests.

hcyezhang committed:
style: pre-commit format

Signed-off-by: Ye Zhang <zhysishu@gmail.com>

rebase in progress; onto 82d03ca97

hcyezhang commented on tensorrt_llm/executor/proxy.py:
```diff
@@ -178,6 +185,16 @@ def process_res(res):
+            # log metrics to prometheus when response is finished and return_perf_metrics is on
+            if isinstance(res, ResponseWrapper):
+                if res._response and res._response.result and res._response.result.is_final:
+                    self._results[client_id]._handle_response(res)
```
(pls correct me if I'm wrong) It seems that `_handle_response` is called by the detokenizer - `DetokenizedGenerationResultBase` if I run the server with trt engine w/o post_process workers. ( i.e. the `metrics_dict` hasn't been generated here)
(wouldn't be a problem if `postprocess_workers` is enabled or pytorch backend is used).

LinPoly replied:
`_handle_response` is called by `RequestOutput`. When we run `result = llm.generate_async(...)`, `result` would be `RequestOutput`, then calling stack is like `async for ... in result` -> `GenerationResult.__anext__` -> `GenerationResult._aresult_step` -> `DetokenizedGenerationResultBase._handle_response`

hcyezhang commented on tensorrt_llm/executor/worker.py:
```diff
@@ -1032,3 +1029,20 @@ def _get_req_perf_metrics(
+    logprobs_result = _get_logprobs(worker, response, is_pytorch_backend)
+    req_perf_metrics = _get_metrics_dict(response)
+    if logprobs_result or req_perf_metrics:
+        if hasattr(response, 'result') and hasattr(response.result, "_result"):
```
I added adaptations here for pytorch engine. (deserialization is needed as here the `_result` is in binary format)

LinPoly replied:
Ah seems like I got your wrong, based on the current implementation, deserialization maybe inevitable, since you choose to construct the `metrics_dict` with `_get_metrics_dict` in the  `GenerationExecutorWorker` side, before responses dispatched, so the result is not deserialized yet. Not sure if it is feasible to move `metrics_dict` construction in `result.py`, after this line, https://github.com/NVIDIA/TensorRT-LLM/blob/2fe9cc08896a020817913d23e1f26ed1222aa719/tensorrt_llm/executor/result.py#L307
but current implementation is fine for me, we can polish this in future PRs. Correct me if I get your wrong again.

hcyezhang committed:
fix: pre-commit B105 false positive

Signed-off-by: Ye Zhang <zhysishu@gmail.com>

LinPoly reviewed (approved):
The test is fine, and we still need to add it to our CI list as following

1. Add it to `test_e2e.py`, like [what we do here](https://github.com/NVIDIA/TensorRT-LLM/blob/e11255e9d0ea1042b33ce7cafae9170ec3f51b1c/tests/integration/defs/test_e2e.py#L1400)
2. Add the above test definition to our CI list, below [this line](https://github.com/NVIDIA/TensorRT-LLM/blob/e11255e9d0ea1042b33ce7cafae9170ec3f51b1c/tests/integration/test_lists/test-db/l0_a10.yml#L16)

Let me know if you have questions about the process, I can help to add it.

hcyezhang committed:
test: add prometheus to integration test

Signed-off-by: Ye Zhang <zhysishu@gmail.com>

hcyezhang commented:
> The test is fine, and we still need to add it to our CI list as following
> 
> 1. Add it to `test_e2e.py`, like [what we do here](https://github.com/NVIDIA/TensorRT-LLM/blob/e11255e9d0ea1042b33ce7cafae9170ec3f51b1c/tests/integration/defs/test_e2e.py#L1400)
> 2. Add the above test definition to our CI list, below [this line](https://github.com/NVIDIA/TensorRT-LLM/blob/e11255e9d0ea1042b33ce7cafae9170ec3f51b1c/tests/integration/test_lists/test-db/l0_a10.yml#L16)
> 
> Let me know if you have questions about the process, I can help to add it.

Thanks, all done. plz have a look when you get a chance :)

hcyezhang committed:
fix: optimize pytorch response handling in _handle_response

Signed-off-by: Ye Zhang <zhysishu@gmail.com>

Superjomn commented:
@FrankD412 @kaiyux A overall question, I remembered trtllm-bench has similar metrics, is it possible to reuse the metrics introduced in this PR?

Superjomn commented:
@hcyezhang  Overall, this looks great, thanks for the contribution. Please address all comments—note that the CodeRabbitAI suggestions are optional, so feel free to resolve any that you consider unfair. Once you’ve made those updates, we’ll help with any remaining minor issues, including running the CI checks.

Superjomn commented on tensorrt_llm/_utils.py:
```diff
@@ -1090,3 +1091,20 @@ def is_multi_device_enable():
+
+def set_prometheus_multiproc_dir() -> object:
+    # Set prometheus multiprocess directory
+    # sglang uses prometheus multiprocess mode
```
Why a "sglang" keyword here? If we reference some code, we may reference a url here.

Superjomn replied:
Ideally, we should not bound to prometheus in the core lib, as there are many other monitoring system. Is it possible to remove this, and give generic results via the public APIs?

FrankD412 replied:
I'm fine with that so long as we do come back to it. I'll defer to @kaiyux and @Superjomn since I'm not an LLM API owner.

Superjomn replied:
@hcyezhang, yeah, lets refine the design in the subsequent PRs, the unittests should protect the further iterations on this feature.

Superjomn commented on tensorrt_llm/executor/proxy.py:
```diff
@@ -60,6 +62,12 @@ def __init__(
         self.workers_started = False
         self.worker_cls = worker_cls
 
+        set_prometheus_multiproc_dir()
```
If this is necessary, how about moving this to trtllm-serve as prometheus is for service deployment? For core API level, it may not be deployed in a cluster.

Shunkangz commented:
/bot run --disable-fail-fast

hcyezhang committed:
fix: move return_perf_metrics arg into sampling_params to keep executor api unchanged

Signed-off-by: Ye Zhang <zhysishu@gmail.com>

hcyezhang committed:
chore: move prometheus dep. to trtllm-serve

Signed-off-by: Ye Zhang <zhysishu@gmail.com>

hcyezhang commented:
> @hcyezhang Overall, this looks great, thanks for the contribution. Please address all comments—note that the CodeRabbitAI suggestions are optional, so feel free to resolve any that you consider unfair. Once you’ve made those updates, we’ll help with any remaining minor issues, including running the CI checks.

Hi @Superjomn, thanks much for reviewing this PR. I think I've addressed all your comments. Could you please give a second pass when you get a chance :)

Superjomn reviewed (approved):
LGTM for the first version, thanks for the contribution. Lets refine it in the following iterations. 
Please address the other comments.

Shunkangz commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #14259](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14259/) [ run ] triggered by Bot

hcyezhang committed:
fix: CI test

Signed-off-by: Ye Zhang <zhysishu@gmail.com>

tensorrt-cicd commented:
[PR_Github #14440](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14440/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #14440](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14440/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #10914](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/10914/) completed with status: 'FAILURE'

hcyezhang committed:
fix: add status field to fix CI tests

Signed-off-by: Ye Zhang <zhysishu@gmail.com>

Commit:
Fix request output reference

Signed-off-by: Shunkang <182541032+Shunkangz@users.noreply.github.co>

Shunkangz commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #14590](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14590/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #14663](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14663/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #14663](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/14663/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #11067](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/11067/) completed with status: 'SUCCESS'
